src/field_theory/krull_topology.lean

```diff
@@ -187,3 +194,84 @@ group_filter_basis.topology (gal_group_basis K L)
 instance (K L : Type*) [field K] [field L] [algebra K L] :
   topological_group (L ≃ₐ[K] L) :=
 group_filter_basis.is_topological_group (gal_group_basis K L)
+
+section krull_t2
+
+
+open_locale topological_space filter
+
+/-- If a subgroup of a topological group has `1` in its interior, then it is open. -/
+lemma open_subgroup_of_nonempty_interior {G : Type*} [group G] [topological_space G]
+[topological_group G] {H : subgroup G} (h_1_int : (1 : G) ∈ interior (H : set G)) :
+is_open (H : set G) :=
+begin
+  have h : 𝓝 1 ≤ 𝓟 (H : set G) :=
+    nhds_le_of_le h_1_int (is_open_interior) (filter.principal_mono.2 interior_subset),
+  rw is_open_iff_nhds,
+  intros g hg,
+  rw (show 𝓝 g = filter.map ⇑(homeomorph.mul_left g) (𝓝 1), by simp),
+  convert filter.map_mono h,
+  simp only [homeomorph.coe_mul_left, filter.map_principal, set.image_mul_left,
+  filter.principal_eq_iff_eq],
+  ext,
+  simp [H.mul_mem_cancel_left (H.inv_mem hg)],
+end
+
+/-- Let `L/E/K` be a tower of fields with `E/K` finite. Then `Gal(L/E)` is an open subgroup of
+  `L ≃ₐ[K] L`. -/
+lemma fixing_subgroup_is_open {K L : Type*} [field K] [field L] [algebra K L]
+{E : intermediate_field K L} (h_findim : finite_dimensional K E) :
+is_open (E.fixing_subgroup : set (L ≃ₐ[K] L)) :=
+begin
+  have h_basis : E.fixing_subgroup.carrier ∈ (gal_group_basis K L) :=
+   ⟨E.fixing_subgroup, ⟨E, h_findim, rfl⟩, rfl⟩,
+  have h_nhd := group_filter_basis.mem_nhds_one (gal_group_basis K L) h_basis,
+  rw mem_nhds_iff at h_nhd,
+  rcases h_nhd with ⟨U, hU_le, hU_open, h1U⟩,
+  exact open_subgroup_of_nonempty_interior ⟨U, ⟨hU_open, hU_le⟩, h1U⟩,
+end
+
+/-- If `L/K` is an algebraic extension, then the Krull topology on `L ≃ₐ[K] L` is Hausdorff. -/
+lemma krull_topology_t2 (K L : Type*) [field K] [field L] [algebra K L] (h_int :
+∀ (x : L), is_integral K x):
```

Comment:
You can use [algebra.is_integral](https://leanprover-community.github.io/mathlib_docs/ring_theory/integral_closure.html#algebra.is_integral) here, that is equal to `h_int` by definition.